protocol/0019-MCAL-margin_calculator.md

```diff
@@ -42,21 +42,125 @@
+
+```math
+margin to remove = current margin * abs(total size of new trades) / abs(position prior to trade)
+```
+
+### Changing Sides
+
+A single order could move a trader from a short to a long position (or vice versa) which either increases or decreases the required margin. When this occurs, the trade should be seen as two steps. One in which the position is reduced to zero (and so margin can be released) followed immediately by one which increases to the new position. (Note that these should *not* be two separate trades, merely modelled as such. The two account movements should occur at once with no other changes allowed between them).
+
+
+### Setting margin mode
+
+When isolated margin mode is enabled,  amount to be transferred is a fraction of the position's notional size that must be specified by the user when enabling isolated margin mode.
+
+The transaction to update/change margin mode can be included in a batch transaction in order to allow updates when placing an order. 
+
+When in isolated margin mode, it is possible to request to both increase or decrease the margin factor setting:
+ - The protocol will attempt to set the funds within the margin account equal to `average entry price * current position * new margin factor`. This value must be above the `initial margin` for the current position or the transaction will be rejected.
+   - If this is less than the balance currently in the margin account, the difference will be moved from the margin account to the general account
+   - If this is larger than the balance currently in the margin account, the difference will be moved from the general account to the margin account. If there are not enough funds to complete this transfer, the transaction will be rejected and no change of margin factor will occur.
+
+When switching to isolated margin mode, the following steps will be taken:
+  1. For any active position, calculate `average entry price * abs(position) * margin factor`. Calculate the amount of funds which will be added to, or subtracted from, the general account in order to do this. If additional funds must be added which are not available, reject the transaction immediately.
+  2. For any active orders, calculate the quantity `limit price * remaining size * margin factor` which needs to be placed in the order margin account. Add this amount to the difference calculated in step 1. If this amount is less than or equal to the amount in the general account, perform the transfers (first move funds into/out of margin account, then move funds into the order margin account). If there are insufficient funds, reject the transaction.
+  3. Move account to isolated margin mode on this market
+
+When switching from isolated margin mode to cross margin mode, the following steps will be taken:
+   1. Any funds in the order margin account will be moved to the cross margin account.
```

Comment:
Does this imply that we're adding two accounts per party per market i.e. 
- isolated margin account and
- order margin account? 

And do I understand correctly that the current margin account (cross margin account stays)?